Login screen

Ocosmee opened the pull request:
# Description
Create view login and components:

 - Email component. 
 - Password component
 - Show password 
 - Login component
 - Sign in with Google component

 # Testing
 Help me how can I test or look at the changes.
- Write the user email and password.
- Click eyes password to show password
 
  ## Screenshots 
If applicable, include screenshots of the results or screenshots that help to see changes


![image](https://user-images.githubusercontent.com/59522204/97736982-06709e00-1aa2-11eb-91da-413f21e79b21.png)
![image (1)](https://user-images.githubusercontent.com/59522204/97737018-11c3c980-1aa2-11eb-94d0-1956a14c00e4.png)
![image (2)](https://user-images.githubusercontent.com/59522204/97737032-16887d80-1aa2-11eb-99c9-e98ab9a034e9.png)

![Captura de Pantalla 2020-10-30 a la(s) 11 12 19](https://user-images.githubusercontent.com/59522204/97737043-1be5c800-1aa2-11eb-9ae5-bff94db5a47c.png)
![Captura de Pantalla 2020-10-30 a la(s) 11 12 12](https://user-images.githubusercontent.com/59522204/97737052-1e482200-1aa2-11eb-831d-dc492801dc7c.png)
![Captura de Pantalla 2020-10-30 a la(s) 11 08 17](https://user-images.githubusercontent.com/59522204/97737067-21dba900-1aa2-11eb-8239-461c3079b014.png)

alcamk committed:
Log in styling details

Ing-Heber committed:
Adjust log in view

alcamk committed:
eye touchable icon

develiasdaniel commented on Reservaciones/Screens/Login.js:
```diff
@@ -1,13 +1,14 @@
 
 const Login = () => {
-   
+   const [isHidden, setIsHidden] = useState(true);
```
Here could be more descriptive like "isHiddenPassword"

Ocosmee committed:
Create set up to conexion of sign in with google and firebase

develiasdaniel reviewed (dismissed):
Design is nice 
Very cool

luisarmandodv commented on Reservaciones/Screens/Login.js:
```diff
@@ -1,13 +1,71 @@
 import styleLogin from '../Styles/LoginStyle';
 import { GoogleSocialButton } from "react-native-social-buttons";
-import { View, Text, TextInput, TouchableOpacity, Linking} from 'react-native';
+import { View, Text, TextInput, TouchableOpacity, Linking, Button} from 'react-native';
+import { FontAwesomeIcon } from '@fortawesome/react-native-fontawesome'
+import { faEye } from '@fortawesome/free-solid-svg-icons'
+import auth from '@react-native-firebase/auth';
+import { GoogleSignin } from '@react-native-community/google-signin';
```
dependency modules should be imported before local importing

luisarmandodv commented on Reservaciones/Screens/Login.js:
```diff
@@ -1,13 +1,71 @@
+import { GoogleSignin } from '@react-native-community/google-signin';
 
+GoogleSignin.configure({
+  webClientId: '346661789891-e6taibn68bvqogs5h93gs9bgdbt3utlp.apps.googleusercontent.com',
```
This is not a must, but it is recommendable to use `react-native-config` in order to handle tokens like this line

luisarmandodv commented on Reservaciones/Screens/Login.js:
```diff
@@ -1,13 +1,71 @@
+});
 
+async function onGoogleButtonPress() {
+   // Get the users ID token
+   const { idToken } = await GoogleSignin.signIn();
+ 
+   // Create a Google credential with the token
+   const googleCredential = auth.GoogleAuthProvider.credential(idToken);
+ 
+   // Sign-in the user with the credential
+   return auth().signInWithCredential(googleCredential);
```
Comments and spaces feel unnecessary.
```suggestion
   const { idToken } = await GoogleSignin.signIn();
   const googleCredential = auth.GoogleAuthProvider.credential(idToken);
 
   return auth().signInWithCredential(googleCredential);
```

luisarmandodv commented on Reservaciones/Screens/Login.js:
```diff
@@ -1,13 +1,71 @@
+ }
 
+function LoginApp() {
+   // Set an initializing state whilst Firebase connects
+   const [initializing, setInitializing] = useState(true);
+   const [user, setUser] = useState();
+ 
+   // Handle user state changes
```
It feels like the code self-explain the purpose of those comments
```suggestion
   const [initializing, setInitializing] = useState(true);
   const [user, setUser] = useState();
 
```

luisarmandodv commented on Reservaciones/Screens/Login.js:
```diff
@@ -1,13 +1,71 @@
+   // Handle user state changes
+   function onAuthStateChanged(user) {
+     setUser(user);
+     if (initializing) setInitializing(false);
```
This is not wrong, and there is no need to be changed, but you could use this approach too
```suggestion
     initializing || setInitializing(false);
```

luisarmandodv commented on Reservaciones/Screens/Login.js:
```diff
@@ -1,13 +1,71 @@
+   const [isHidden, setIsHidden] = useState(true);
+
+
+  cerrarsesion = () => {
```
Arrow function name shouldn't be spanish.

luisarmandodv commented on Reservaciones/Screens/Login.js:
```diff
@@ -17,25 +75,41 @@ const Login = () => {
-            </Text>                
+            </Text>
+            <LoginApp/>
+            <Button
+            title= 'Cerrar sesion'
+            onPress = {() => cerrarsesion()}/>
```
```suggestion
            <Button
              title= 'Cerrar sesion'
              onPress = {() => cerrarsesion()}/>
```

luisarmandodv commented on Reservaciones/Screens/Login.js:
```diff
@@ -17,25 +75,41 @@ const Login = () => {
             </TouchableOpacity>
             <Text style={{marginTop: 20}}>
                <Text style={{color: '#B6B7BA'}}>Don´t have an account? </Text>
                <Text style={{color: '#5974F5', textDecorationLine: 'underline'}} 
 		         onPress={ ()=> Linking.openURL('https://www.google.com/') }>Sign Up</Text>    
```
indentation here is not consistent with the spacing used among all the code

luisarmandodv reviewed (changes requested):
All components should have `PropTypes` defined in them.

https://es.reactjs.org/docs/typechecking-with-proptypes.html

JordiGN reviewed (dismissed):
It's cool overall, and I see you start doing the consumption of the FirebaseServices, that's nice, my only suggestions, when you write in the inputs it could be nice to have a different color in the border

Ing-Heber committed:
Code corrections from the code review

alcamk committed:
firebase login functionality initial version

Ocosmee committed:
Add styles to inputs components

alcamk committed:
fixed color login button

alcamk committed:
Merge branch 'loginScreen' of https://github.com/Brightcoders-Bootcamps/kata-05-reservacion-vuelo-team-01-kata-05 into loginScreen

Ocosmee committed:
Validation message

Ing-Heber committed:
Fixed  buttons

Ocosmee committed:
Added Loader component

tovl1 reviewed (dismissed):
Good!

luisarmandodv commented on Reservaciones/Screens/Loader.js:
```diff
@@ -0,0 +1,56 @@
+const Loader = props => {
+  const {
+    loading,
+    ...attributes
```
isn't this causing an error? `attributes` is not being defined anywhere

luisarmandodv commented on Reservaciones/Screens/Loader.js:
```diff
@@ -0,0 +1,56 @@
+  } = props;
+
+  return (
+    <Modal
+      transparent={true}
+      animationType={'none'}
+      visible={loading}
+      onRequestClose={() => {console.log('close modal')}}>
+      <View style={styles.modalBackground}>
+        <View style={styles.activityIndicatorWrapper}>
+            <View>
+                <ActivityIndicator
+                    size='large'
+                    color='#5C6EF8'
+                    animating={loading} />
+                    <Text style={{color: '#5C6EF8', marginTop: 25}}>Logging In...</Text>
+            </View>
+        </View>
+      </View>
+    </Modal>
```
indentation here is not consistent, be sure to use two spaces o four, but they need to be consistent among all project. I recommend you to check the `Prettier` tool -> https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode

luisarmandodv commented on Reservaciones/Screens/Loader.js:
```diff
@@ -0,0 +1,56 @@
+  }
+});
+
+export default Loader;
```
```suggestion
export default Loader;

```

LeyaimMagma replied:
Add space here

luisarmandodv commented on Reservaciones/Screens/Login.js:
```diff
@@ -1,44 +1,271 @@
 
+GoogleSignin.configure({
+  webClientId:
+    '346661789891-e6taibn68bvqogs5h93gs9bgdbt3utlp.apps.googleusercontent.com',
```
consider using `react-native-config` in a near future to replace this.

luisarmandodv commented on Reservaciones/Screens/Login.js:
```diff
@@ -1,44 +1,271 @@
+        });
+
+        await auth().signInWithEmailAndPassword(email, password);
+        // alert('entro por email y pass');
+        //await auth().onAuthStateChanged(user => {  alert(user.email);  })
```
```suggestion
```

luisarmandodv commented on Reservaciones/Screens/Login.js:
```diff
@@ -1,44 +1,271 @@
+        }, 2500);
+      }
+    } catch (error) {
+      let errorMessage = error.message.toString(error);
+      this.setState({errorLogin: errorMessage});
```
use constants if the value is not modified
```suggestion
      const errorMessage = error.message.toString(error);
      this.setState({errorLogin: errorMessage});
```

luisarmandodv commented on Reservaciones/Screens/Login.js:
```diff
@@ -1,44 +1,271 @@
+            <View>
+              <Text style={styleLogin.TextTittle}>Log In</Text>
+              <Text style={styleLogin.InputTittle}>Email * </Text>
+
```
I don't consider break-lines inside JSX a good practice
```suggestion
```

luisarmandodv replied:
several cases on this component, please review them all

luisarmandodv commented on Reservaciones/Screens/Login.js:
```diff
@@ -1,44 +1,271 @@
+                  {this.state.errorLogin}
+                </Text>
+              </View>
+
```
```suggestion
```

luisarmandodv commented on Reservaciones/Screens/Login.js:
```diff
@@ -1,44 +1,271 @@
+                keyboardType={'email-address'}
+                onChangeText={(val) => this.setState({userEmail: val})}
+              />
+
```
```suggestion
```

luisarmandodv commented on Reservaciones/Screens/Login.js:
```diff
@@ -1,44 +1,271 @@
+  }
+
+   
+  render() {
+    return (
       <View style={styleLogin.Father}>
```
I'm concerned about the number of JSX lines this screen has, please review if it is possible to refactor it into smaller reusable components.

luisarmandodv replied:
still same comment 👆 , Please review if there is any chance to refactor it into smaller components on different files

luisarmandodv commented on Reservaciones/Screens/Login.js:
```diff
@@ -1,44 +1,271 @@
+      })
+      .catch((error) => {
+        if (error.code === 'auth/email-already-in-use') {
+          console.log('That email address is already in use!');
```
```suggestion
```

luisarmandodv commented on Reservaciones/Screens/Login.js:
```diff
@@ -1,44 +1,271 @@
+          console.log('That email address is already in use!');
+        }
+        if (error.code === 'auth/invalid-email') {
+          console.log('That email address is invalid!');
```
```suggestion
```

luisarmandodv commented on Reservaciones/Screens/Login.js:
```diff
@@ -1,44 +1,271 @@
+        if (error.code === 'auth/invalid-email') {
+          console.log('That email address is invalid!');
+        }
+        console.error(error);
```
```suggestion
```

luisarmandodv commented on Reservaciones/Screens/Login.js:
```diff
@@ -1,44 +1,271 @@
+  );
+}
+
+class Login extends React.Component {
```
I see you have `LoginApp` hook and `Login` class. I think it would be better to make a file by component.

LeyaimMagma commented on Reservaciones/Screens/Loader.js:
```diff
@@ -0,0 +1,56 @@
+            <View>
+                <ActivityIndicator
+                    size='large'
+                    color='#5C6EF8'
```
Create a color constant file and all the colors that you gonna use

luisarmandodv commented:
@Ocosmee Be aware of comments, `console.logs`, identation and spacing. Try using `Prettier`.

LeyaimMagma commented on Reservaciones/Screens/Login.js:
```diff
@@ -1,44 +1,271 @@
-               <TextInput secureTextEntry={true}></TextInput>
+          <View style={styleLogin.eyeContainer}>
+            <View>
+              <Text style={styleLogin.TextTittle}>Log In</Text>
+              <Text style={styleLogin.InputTittle}>Email * </Text>
```
Create a constant file to add all the text

JordiGN reviewed (dismissed):
Well done, I think it was a good way to name the files, also I notice the use of Google Sign in, that's cool, keep working on the observation the mentors have told you. Nice job, keep doing that way

Ing-Heber committed:
Applied mentor suggestions

luisarmandodv commented on Reservaciones/App.js:
```diff
@@ -1,20 +1,16 @@
-import { NavigationContainer } from '@react-navigation/native';
+import {NavigationContainer} from '@react-navigation/native';
 import LoginView from './Screens/Login';
 
```
```suggestion
```

luisarmandodv commented on Reservaciones/Screens/Loader.js:
```diff
@@ -0,0 +1,53 @@
+              animating={loading}
+            />
+            <Text style={{color: Colors.blue, marginTop: 25}}>
+              Logging In...
```
This text should be in the `Text.js` file

luisarmandodv commented on Reservaciones/Screens/Loader.js:
```diff
@@ -0,0 +1,53 @@
+      animationType={'none'}
+      visible={loading}
+      onRequestClose={() => {
+        console.log('close modal');
```
```suggestion
```

luisarmandodv commented on Reservaciones/Screens/Login.js:
```diff
@@ -1,44 +1,195 @@
+      if (email.trim() === '') {
+        alert('Indtroduzca un email por favor');
+      } else if (password.trim() === '') {
+        alert('Indtroduzca un password por favor');
```
```suggestion
        alert('Introduzca un password por favor');
```

luisarmandodv commented on Reservaciones/Screens/Login.js:
```diff
@@ -1,44 +1,195 @@
+      }
+    } catch (error) {
+      const errorMessage = error.message.toString(error);
+      this.setState({errorLogin: errorMessage});
+      this.setState({
+        loading: false,
+      });
```
```suggestion
      this.setState({
        errorLogin: errorMessage,
        loading: false,
      });
```

luisarmandodv commented on Reservaciones/Screens/Login.js:
```diff
@@ -1,44 +1,195 @@
+            />
+            <Text style={styleLogin.loginText}>{Texts.login_with_google}</Text>
+          </TouchableOpacity>
+
```
```suggestion
```

luisarmandodv commented on Reservaciones/Screens/Login.js:
```diff
@@ -1,44 +1,195 @@
+          <Text style={{textAlign: 'center', color: Colors.gray, margin: 10}}>
+            {Texts.or}
+          </Text>
+
```
```suggestion
```

luisarmandodv commented on Reservaciones/Screens/Login.js:
```diff
@@ -1,44 +1,195 @@
+            }>
+            <Text style={styleLogin.loginText}>{Texts.login}</Text>
+          </TouchableOpacity>
+
```
```suggestion
```

luisarmandodv commented on Reservaciones/Styles/LoginStyle.js:
```diff
@@ -1,93 +1,146 @@
+    paddingBottom: 10,
+    paddingLeft: 0,
+    backgroundColor: Colors.white,
+    color: '#424242',
```
could this color be in the Colors.js file?

luisarmandodv commented on Reservaciones/Screens/Login.js:
```diff
@@ -1,44 +1,195 @@
+                : styleLogin.GoogleLoginButton,
+            ]}
+            onPress={() =>
+              onGoogleButtonPress().then(() =>
+                console.log('Signed in with Google!'),
+              )
```
I recommendd the use of `async` `await` instead of `.then()`.

luisarmandodv reviewed (changes requested):
All components should have PropTypes, take a look at this documentation:
https://es.reactjs.org/docs/typechecking-with-proptypes.html

Ocosmee committed:
Implement Logn In suggestions

Ocosmee committed:
Create components for Button and InputText

Ocosmee committed:
Added PropTypes on the project

Ocosmee committed:
Adding navigation between screns

From Login to Sign Up

alcamk committed:
Login view style and navigation functionality

alcamk committed:
Login view style and navigation functionality between login and singup

alcamk committed:
fixing navigation problems

alcamk committed:
Login screen

"Login" button needs to adjust properties for correct functionality.

alcamk committed:
Login Screen

Complete funcionality of buttons, text inputs, navigation and login.

luisarmandodv commented on Reservaciones/Styles/signup.js:
```diff
@@ -0,0 +1,22 @@
+    txtItm: {
+        marginTop: 8,
+    }
+
```
```suggestion
```

luisarmandodv commented on Reservaciones/Styles/signup.js:
```diff
@@ -0,0 +1,22 @@
+
+});
+
+export default SignupStyle;
```
```suggestion
export default SignupStyle;

```

luisarmandodv commented on Reservaciones/Styles/checkbox.js:
```diff
@@ -0,0 +1,21 @@
+import {StyleSheet} from 'react-native';
+
+const checkboxStyle = StyleSheet.create({
+    container: {
+        marginBottom: 20,
+        width: 30,
+        height: 30,
+        flexDirection: "row",
+    },
+    item: {
+        marginRight: 10,
+    },
+    txtChkbx: {
+        color: '#AFB2B7',
+        width: '100%'
+
+    }
+
```
```suggestion
    }
```

luisarmandodv replied:
You are using 4 spaces of indentation here, but on other components/files you are using 2 spaces of indentation.

luisarmandodv commented on Reservaciones/Styles/signup.js:
```diff
@@ -0,0 +1,22 @@
+    },
+    txtItm: {
+        marginTop: 8,
+    }
```
You are using 4 spaces of indentation here, but on other components/files you are using 2 spaces of indentation.

luisarmandodv commented on Reservaciones/Styles/LoginStyle.js:
```diff
@@ -1,93 +1,130 @@
+    borderRadius: 10,
+    borderWidth: 1,
+    borderColor: Colors.white,
+    height: 47,
+    justifyContent: 'center',
+    marginRight: 40,
+    marginLeft: 0,
+    marginTop: 10,
+    width: '100%'
+  },
+  loginText: {
+    color: Colors.white,
+    textAlign: 'center',
+    fontWeight: 'bold',
+    paddingBottom: 0,
+    fontSize: 17,
+  },
+  searchSection: {
+    flex: 1,
+    flexDirection: 'row',
+    justifyContent: 'center',
+    alignItems: 'center',
+    backgroundColor: Colors.white,
+  },
+  searchIcon: {
+    padding: 10,
+  },
+  eyeIcon: {
+    paddingTop: 44, 
+    paddingRight: 18,
+    position: 'absolute',
+    alignSelf: 'flex-end',
+  },
+  errorLogin: {
+    color: Colors.red,
+  },
+  errorPass: {
+    borderColor: Colors.red,
+  },
+});
```
I see some of those style names begin with lowercase and other ones with uppercase. You need to be consistent with the names and use a single way. I recommend the use of `lowerCase`, for example:
`GoogleLoginButton` -> `googleLoginButton`

luisarmandodv commented on Reservaciones/Screens/SignUp.js:
```diff
@@ -0,0 +1,175 @@
+            <Text style={styleLogin.TextTittle}>{Texts.logIn}</Text>
+                <View>
+                <TextField
+                    title={Texts.email}
+                    nameHandlerFocus={this.state.Vemail}
+                    Hfocus={() => this.handlerFocus(Texts.Pemail)}
+                    HBlur={() => this.handlerBlur(Texts.Pemail, this.state.userEmail)}
+                    campo= 'userEmail'
+                    keyboard = {false}
+                    errorLogin=''
+                    changeText={this.handleChangeText}
+                  />
```
four indentation spaces here, which is not consistent with the 2 spaces you are using among all the component

luisarmandodv commented on Reservaciones/Screens/Login.js:
```diff
@@ -1,44 +1,177 @@
+        </View>
+        <View style={styleLogin.ButomsArea}>
+          <ButtonAction 
+                stateComponent = {this.state}
+                title={Texts.login}
+                imageRequired={false}
+                Press = {() => this.SignInWithEmailPass(this.state.userEmail,this.state.userPass)}
```
four indentation spaces here, which is not consistent with the 2 spaces you are using among all the component

luisarmandodv commented on Reservaciones/Screens/Login.js:
```diff
@@ -1,44 +1,177 @@
+          <View style={styleLogin.eyeContainer}>
+            <View>
+            <Text style={styleLogin.TextTittle}>{Texts.logIn}</Text>
+                <View>
+                <TextField
+                    title={Texts.email}
+                    nameHandlerFocus={this.state.Vemail}
+                    Hfocus={() => this.handlerFocus(Texts.Pemail)}
+                    HBlur={() => this.handlerBlur(Texts.Pemail, this.state.userEmail)}
+                    campo= 'userEmail'
+                    keyboard = {false}
+                    errorLogin=''
+                    changeText={this.handleChangeText}
+                  />
+                </View>
+              <View style={styleLogin.eyeContainer}>
+                <View>
+                  <TextField
+                    title={Texts.pass}
+                    nameHandlerFocus={this.state.Vpassword}
+                    Hfocus={() => this.handlerFocus(Texts.Ppassword)}
+                    HBlur={() => this.handlerBlur(Texts.Ppassword,this.state.userPass)}
+                    campo='userPass'
+                    keyboard = {this.state.isPasswordHidden}
+                    errorLogin={this.state.errorLogin}
+                    changeText={this.handleChangeText}
+                  />
```
four indentation spaces here, which is not consistent with the 2 spaces you are using among all the component

luisarmandodv commented on Reservaciones/Screens/Loader.js:
```diff
@@ -0,0 +1,62 @@
+import PropTypes from 'prop-types';
+import {Colors} from '../Styles/Colors';
+import {Texts} from '../ContentText/Texts';
+
```
```suggestion
```

luisarmandodv commented on Reservaciones/ContentText/SignupTexts.js:
```diff
@@ -0,0 +1,21 @@
+    star: '+',
+    checkbox2: 'Suscribe for select product updates.',
+  };
+  
```
```suggestion
```

luisarmandodv commented on Reservaciones/Components/checkbox.js:
```diff
@@ -0,0 +1,33 @@
+
+export default ComCheckbox;
+
+
```
```suggestion
```

luisarmandodv commented on Reservaciones/Components/checkbox.js:
```diff
@@ -0,0 +1,33 @@
+
+
+    return (
+        <>
+            <View style={checkboxStyle.container}>
+                <View style={checkboxStyle.item}>
+                    <CheckBox
+                        disabled={false}
+                        value={toggleCheckBox}
+                        boxType={'square'}
+                        onCheckColor={'white'}
+                        onFillColor={'#5C6EF8'}
+                        onTintColor={'#5C6EF8'}
+                        onValueChange={(newValue) => setToggleCheckBox(newValue)}
+                    />
+                </View>
+            </View>
+        </>
```
four indentation spaces here, which is not consistent with the 2 spaces you are using among all the project

luisarmandodv commented on Reservaciones/Components/InputField.js:
```diff
@@ -0,0 +1,45 @@
+import {Texts} from '../ContentText/Texts';
+
+
+
```
```suggestion
```

luisarmandodv commented on Reservaciones/Components/InputField.js:
```diff
@@ -0,0 +1,45 @@
+        }
+      };
+
+    <View>
+        <Text style={styleLogin.InputTittle}>{Texts.email}</Text>
+        <TextInput
+        style={[
+            styleLogin.InputText,
+            this.state.nameInputOneFocus
+            ? styleLogin.textInputFocus
+            : styleLogin.InputText,
+        ]}
+        onFocus={() => this.handlerFocus('nameInputOneFocus')}
+        onBlur={() => this.handlerBlur('nameInputOneFocus')}
+        keyboardType={'email-address'}
+        onChangeText={(val) => this.setState({userEmail: val})}
+        />
+    </View>
```
indentation here can improve

alcamk committed:
Update Reservaciones/Styles/signup.js

Co-authored-by: Armando Duran <luisarmandodv@gmail.com>

Ocosmee committed:
Changes suggestions done

luisarmandodv commented on Reservaciones/Components/Button.js:
```diff
@@ -0,0 +1,42 @@
+    super(props);
+  }
+  render() {
+    const {stateComponent,title,imageRequired, Press} = this.props;
```
`stateComponent` is not defined on PropTypes

luisarmandodv commented on Reservaciones/Components/InputField.js:
```diff
@@ -0,0 +1,40 @@
+import { View,  Text, TextInput } from "react-native";
+import React, {useState, useEffect, Component} from 'react';
```
```suggestion
import React, {useState, useEffect} from 'react';
```

luisarmandodv commented on Reservaciones/Components/TextField.js:
```diff
@@ -0,0 +1,42 @@
+    super(props);
+  }
+  render() {
+    const {nameHandlerFocus, Hfocus, HBlur, campo, changeText, title, keyboard, errorLogin} = this.props;
```
`nameHandlerFocus` not defined on PropTypes

luisarmandodv replied:
what happened with this? @Ocosmee

Ocosmee replied:
I'm sorry, I didn't know how to do it, but I made the correction

luisarmandodv commented on Reservaciones/Components/TextField.js:
```diff
@@ -0,0 +1,42 @@
+import styleLogin from '../Styles/LoginStyle';
+
+class TextField extends Component {
+  constructor(props) {
+    super(props);
+  }
```
`constructor` not necessary in this component

luisarmandodv commented on Reservaciones/Screens/Loader.js:
```diff
@@ -0,0 +1,60 @@
+
+Loader.propTypes = {
+  loading: PropTypes.bool,
+  transparent: PropTypes.bool,
+  animationType: PropTypes.string,
+  size : PropTypes.string,
+  color: PropTypes.string,
+  animating:  PropTypes.bool,
```
those are not props of this component

Ocosmee committed:
Changes suggestions

Ocosmee committed:
Change suggestion nameHandlerFocus

JordiGN reviewed (commented):
Great, I see you already implemented Firebase. A recommendation is the use of prettier on every file you have worked on, at the end of the session, just to make sure it has the right format

tovl1 reviewed (dismissed):
Its good

alcamk committed:
Merging changes

Accept and delete code extensions to upgrade the last feature pushed to the login screen branch, and fixing merging problems

alcamk committed:
Merge branch 'sand-box' into loginScreen

develiasdaniel reviewed (dismissed):
Good job team!

OwenRamire commented on Reservaciones/Components/checkbox.js:
```diff
@@ -0,0 +1,28 @@
+const  ComCheckbox = (props) => {
+  const [toggleCheckBox, setToggleCheckBox] = useState(false)
+  return (
+    <>
```
Do you really need this?

OwenRamire commented on Reservaciones/Screens/Loader.js:
```diff
@@ -0,0 +1,55 @@
+import React, {Component} from 'react';
+import {StyleSheet, View, Modal, ActivityIndicator, Text} from 'react-native';
```
```suggestion
import {
  StyleSheet,
  View,
  Modal,
  ActivityIndicator,
  Text} from 'react-native';
```

OwenRamire commented on Reservaciones/Screens/Login.js:
```diff
@@ -1,41 +1,183 @@
-import React from 'react';
+import React, {useState, useEffect, Component} from 'react';
```
`Hooks` don't work in a class component
```suggestion
import React, {Component} from 'react';
```

OwenRamire commented on Reservaciones/Screens/SignUp.js:
```diff
@@ -0,0 +1,175 @@
+          <View style={styleLogin.eyeContainer}>
+            <View>
+              <Text style={styleLogin.textTittle}>{Texts.logIn}</Text>
+                <View>
+                  <TextField
+                    title={Texts.email}
+                    nameHandlerFocus={this.state.Vemail}
+                    Hfocus={() => this.handlerFocus(Texts.Pemail)}
+                    HBlur={() => this.handlerBlur(Texts.Pemail, this.state.userEmail)}
+                    campo= 'userEmail'
+                    keyboard = {false}
+                    errorLogin=''
+                    changeText={this.handleChangeText}
+                  />
+                </View>
+              <View style={styleLogin.eyeContainer}>
+                <View>
+                  <TextField
+                    title={Texts.pass}
+                    nameHandlerFocus={this.state.Vpassword}
+                    Hfocus={() => this.handlerFocus(Texts.Ppassword)}
+                    HBlur={() => this.handlerBlur(Texts.Ppassword,this.state.userPass)}
+                    campo='userPass'
+                    keyboard = {this.state.isPasswordHidden}
+                    errorLogin={this.state.errorLogin}
+                    changeText={this.handleChangeText}
+                  />
```
You can refactor this in a function
https://reactjs.org/docs/lists-and-keys.html

Ocosmee committed:
Added Sign Up validation

Ocosmee committed:
Add screen myflights

Ing-Heber committed:
Fixed styles in login, signup and myflights
